dev/ci/runtype/runtype.go

```diff
@@ -92,6 +94,13 @@ func (t RunType) Matcher() *RunTypeMatcher {
 			},
 		}
 
+	case Targeted:
+		return &RunTypeMatcher{
+			Branch:                    "targeted/",
+			BranchArgumentRequired:    true,
+			BranchArgumentDescription: `A substring of a valid step label, with space replaced by dashes: "backend integration" will match ":chains: Backend integration tests"`,
```

Comment:
If we start adding docstrings here, let's make it clearer what is associated with what using a struct:

```suggestion
			BranchArgument:  RunTypeBranchArgument{
				Required: true,
				Description: `A substring of a valid step label, with space replaced by dashes: "backend integration" will match ":chains: Backend integration tests"`,
			}
```